crates/bevy_scene/src/scene_loader.rs

```diff
@@ -35,7 +35,17 @@ impl AssetLoader for SceneLoader {
             let scene_deserializer = SceneDeserializer {
                 type_registry: &self.type_registry.read(),
             };
-            let scene = scene_deserializer.deserialize(&mut deserializer)?;
+            let scene = scene_deserializer
+                .deserialize(&mut deserializer)
+                .map_err(|e| {
+                    let span_error = deserializer.span_error(e);
+                    anyhow!(
```

Comment:
I'd really like for us to reduce our use of `anyhow`. IMO it's a crate that end-users should be using, not the libraries they're depending on. With that said, this isn't a breaking change from what we're already doing, so I won't block on this.

Reply:
Yeah I was just using anyhow here because it was already the mechanism being used (The `?` was converting it to an anyhow Result). Normally for error handling in libs you would have an error enum that would have variants for the possible failure states and that would hold the context needed to display this error. You would construct that variant here instead. Should be easy enough to migrate this to that if/when we make a switch away from anyhow.